### PR TITLE
Added KeepsCommanderTest

### PR DESCRIPTION
**Pull request by ahsu1230**
```
test only tests keep exports for now
export keeps backend functionality moved from KeepsController to KeepsCommander
```

changed a few other minor things

### CODE REVIEW

**Comment by ahsu1230** [comment]
@martinraison @eishay @andrewconner @stkem

**Comment by ahsu1230** [comment]
@leogrim @lawzlo

**Comment by leogrim** [comment]
Sweet, good job minimizing your Guice dependencies here :)

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala [hunk 1]
```diff
@@ -625,4 +625,30 @@ class KeepsCommander @Inject() (
...
+        s""" TAGS="${tags.replace("&","&amp;").replace("\"","")}""""
+      } getOrElse ""
+      val date = keep.createdAt.getMillis()/1000
+      val line =
+        s"""<DT><A HREF="${keep.url}" ADD_DATE="${date}"${tagString}>${title.replace("&","&amp;")}</A>"""
+      line
+    }
+    before + keepExports.map(createExport).mkString("\n") + after
+  }
+}
```
Comment (martinraison) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala]: new line (I know, it's annoying :) )

File: kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala [hunk 2]
```diff
@@ -349,4 +349,4 @@ class KeepRepoImpl @Inject() (
       group by url order by k.id desc"""
     sql_query.as[(DateTime, Option[String], String, Option[String])].list.map{ case (created_at, title, url, tags) => KeepExport(created_at, title, url, tags)}
   }
-}
+}
```
Comment (martinraison) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala]: :)

File: kifi-backend/modules/shoebox/test/com/keepit/commanders/KeepsCommanderTest.scala [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          keepToCollectionRepo.save(KeepToCollection(keepId = keep1.id.get, collectionId = col1.id.get))
+        }
+
+        db.readOnly { implicit s =>
+          val keepExports = keepRepo.getKeepExports(Id[User](1))
+          keepExports.length === 3
+          keepExports(0) === KeepExport(title = Some("k3"), createdAt = t1.plusMinutes(6), url = site3)
+          keepExports(1) === KeepExport(title = Some("k2"), createdAt = t1.plusMinutes(9), url = site2)
+          keepExports(2) === KeepExport(title = Some("k1"), createdAt = t1.plusMinutes(3), url = site1, tags = Some("t1"))
+        }
```
Comment (martinraison) [inline kifi-backend/modules/shoebox/test/com/keepit/commanders/KeepsCommanderTest.scala]: not very important for a test, but you generally want to keep your database sessions as short as possible. Here you could have

```
val keepExports = db.readOnly { implicit s => keepRepo.getKeepExports(Id[User](1))}
keepExports.length === 3
keepExports(0) === KeepExport(title = Some("k3"), createdAt = t1.plusMinutes(6), url = site3)
keepExports(1) === KeepExport(title = Some("k2"), createdAt = t1.plusMinutes(9), url = site2)
keepExports(2) === KeepExport(title = Some("k1"), createdAt = t1.plusMinutes(3), url = site1, tags = Some("t1"))
```

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala (hunk 1 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala]: Oh! So we need a new line after. I thought we didn't want them. gotcha

File: kifi-backend/modules/shoebox/test/com/keepit/commanders/KeepsCommanderTest.scala [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+             |<DL>
+             |<DT><A HREF="http://www.hi.com11" ADD_DATE="$seconds0" TAGS="tagA">title 1&amp;1</A>
+             |<DT><A HREF="http://www.hi.com21" ADD_DATE="$seconds1" TAGS="tagA,tagB&amp;tagC">title 21</A>
+             |<DT><A HREF="http://www.hi.com31" ADD_DATE="$seconds2">title 31</A>
+             |</DL>""".stripMargin
+        result must equalTo(expected)
+      }
+    }
+  }
+}
```
Comment (martinraison) [inline kifi-backend/modules/shoebox/test/com/keepit/commanders/KeepsCommanderTest.scala]: (:

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala (hunk 1 above)
Comment (martinraison) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala]: Yes, it comes in handy in some situations (http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline for example)

**Comment by martinraison** [comment]
Nicely done! Feel free to merge after addressing the feedback.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala]: Note: You can configure any editor to automatically add a final newline to text files on save if it’s missing — google "[editor name] add newline end of file on save".

**Pull request merged**